numpy/_core/src/multiarray/item_selection.c

```diff
@@ -1455,7 +1491,9 @@ _new_argsortlike(PyArrayObject *op, int axis, PyArray_ArgSortFunc *argsort,
         }
     }
 
-    NPY_BEGIN_THREADS_DESCR(descr);
+    if (!needs_api) {
+        NPY_BEGIN_THREADS;
+    }
```

Comment:
Here we do also need to usethe descriptors, in this case the first (as the second we assume is default intp, I am fine with that). But thecopying must use context->descrs[0]

Reply:
Forget about it, I think I just picked a pretty random place in the argsort loop to comment that the desriptor is important there also. You added that.